### PR TITLE
wr-322: add-prompt-usage-confirmation

### PR DESCRIPTION
**Pull request by canterbery**
image

### CODE REVIEW

File: frontend/src/libs/components/toggle-checkbox/toggle-checkbox.tsx [hunk 1]
```diff
@@ -33,11 +35,14 @@ const ToggleCheckbox = <T extends FieldValues>({
         {...field}
         name={name}
         type="checkbox"
-        id={TOGGLE_CHECKBOX_ID}
+        id={checkBoxId ?? TOGGLE_CHECKBOX_ID}
```
Comment (andanf-e) [inline frontend/src/libs/components/toggle-checkbox/toggle-checkbox.tsx]: Can we use field name for this purpose?

File: frontend/src/pages/article/components/prompt-confirmation/libs/helpers/get-confirmed-prompts.helper.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { type PromptConfirmationFormValues } from '../types/types.js';
+
+const getConfirmedPrompts = (
+  prompt: GeneratedPrompt,
+  confirmedPrompt: PromptConfirmationFormValues,
+): PromptRequestDto | null => {
+  return getGeneratedPromptPayload(
+    Object.entries(prompt).reduce<GeneratedPrompt>((resultPrompt, [key]) => {
+      const castedKey = key as keyof typeof prompt;
+      if (castedKey !== 'genre' && !confirmedPrompt[castedKey]) {
```
Comment (andanf-e) [inline frontend/src/pages/article/components/prompt-confirmation/libs/helpers/get-confirmed-prompts.helper.ts]:
```suggestion
      if (castedKey !== PromptCategory.GENRE && !confirmedPrompt[castedKey]) {
```

File: frontend/src/pages/article/components/prompt-confirmation/prompt-confirmation-form.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import { getConfirmedPrompts } from './libs/helpers/helpers.js';
+import { type PromptConfirmationFormValues } from './libs/types/types.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  submitData: {
+    prompt: GeneratedPrompt;
+    payload: ArticleRequestDto;
+  };
+  handleSubmitType: (value: ValueOf<typeof ArticleSubmitType> | null) => void;
```
Comment (andanf-e) [inline frontend/src/pages/article/components/prompt-confirmation/prompt-confirmation-form.tsx]:
```suggestion
  onSetSubmitType: (value: ValueOf<typeof ArticleSubmitType> | null) => void;
```

File: frontend/src/pages/article/components/prompt-confirmation/prompt-confirmation-form.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  return (
+    <div className={styles.formWrapper}>
+      <form
+        className={styles.innerWrapper}
+        name="ConfirmPromptsForm"
+        onSubmit={handleFormSubmit}
+      >
+        {Object.entries(prompt).map(([key, value]) => {
+          const castedKey = key as keyof typeof prompt;
+          if (castedKey === 'genre') {
```
Comment (andanf-e) [inline frontend/src/pages/article/components/prompt-confirmation/prompt-confirmation-form.tsx]:
```suggestion
          if (castedKey === PromptCategory.GENRE) {
```

File: frontend/src/pages/article/components/prompt-confirmation/libs/helpers/get-confirmed-prompts.helper.ts (hunk 2 above)
Comment (canterbery) [reply frontend/src/pages/article/components/prompt-confirmation/libs/helpers/get-confirmed-prompts.helper.ts]: do we really need it ? 
image

File: frontend/src/libs/components/toggle-checkbox/toggle-checkbox.tsx [hunk 5]
```diff
@@ -33,11 +33,11 @@ const ToggleCheckbox = <T extends FieldValues>({
         {...field}
         name={name}
         type="checkbox"
-        id={TOGGLE_CHECKBOX_ID}
+        id={name ?? TOGGLE_CHECKBOX_ID}
```
Comment (andanf-e) [inline frontend/src/libs/components/toggle-checkbox/toggle-checkbox.tsx]: I guess name is not optional so we can also get rid of nullish coalescing

**Review by andanf-e — DISMISSED** [review]

File: frontend/src/libs/components/toggle-checkbox/toggle-checkbox.tsx (hunk 5 above)
Comment (canterbery) [reply frontend/src/libs/components/toggle-checkbox/toggle-checkbox.tsx]: And that constant as well I guess

**Review by andanf-e — APPROVED** [review]

**Pull request merged**